### PR TITLE
Add a sample dataset MaunaLoa_CO2

### PR DESCRIPTION
**Pull request by willschlitzer**
This adds `@MaunaLoa_CO2.txt` to `load_sample_data()`.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/datasets/samples.py [hunk 1]
```diff
@@ -371,6 +373,21 @@ def _load_notre_dame_topography():
...
 
 
+def _load_maunaloa_co2():
+    """
+    Load a table of CO2 values from Mauna Loa.
+
+    Returns
+    -------
+    data : pandas.DataFrame
+        The data table with columns "data" and "co2_ppm".
```
Comment (weiji14) [inline pygmt/datasets/samples.py:383]:
```suggestion
        The data table with columns "date" and "co2_ppm".
```

File: pygmt/tests/test_datasets_samples.py [hunk 2]
```diff
@@ -173,3 +173,16 @@ def test_earth_relief_holes():
...
+
+def test_maunaloa_co2():
+    """
+    Check that the @MaunaLoa_CO2.txt dataset loads without errors.
+    """
+    data = load_sample_data(name="maunaloa_co2")
+    assert data.shape == (730, 2)
+    summary = data.describe()
+    assert summary.loc["min", "date"] == 1958.2027
+    assert summary.loc["max", "date"] == 2019.3699
```
Comment (weiji14) [inline pygmt/tests/test_datasets_samples.py:186]: Is the `date` column a floating point column? Should we convert it to a proper pandas datetime dtype?

Comment (willschlitzer) [reply pygmt/tests/test_datasets_samples.py:186]: It is a floating point number in the original dataset. My concern with converting to a date time object is that the GMT API won't recognize that input.

Comment (weiji14) [reply pygmt/tests/test_datasets_samples.py:186]: PyGMT does work with datetime I'm sure. See https://www.pygmt.org/v0.6.1/tutorials/advanced/date_time_charts.html?highlight=datetime

**Review by michaelgrund — APPROVED** [review]

File: pygmt/tests/test_datasets_samples.py (hunk 2 above)
Comment (willschlitzer) [reply pygmt/tests/test_datasets_samples.py:186]: @GenericMappingTools/pygmt-maintainers Does anyone else have an input on this? My thought is to maintain the original date type in the remote file as a float, but I don't feel that strongly.

Comment (seisman) [reply pygmt/tests/test_datasets_samples.py:186]: I'm OK with a float-type date.

**Pull request merged**